app/ui/browser-modern/sagas/page-sagas.js

```diff
@@ -49,18 +50,25 @@ export default function*() {
   ];
 }
 
-function* createPageSession({ id, location, options }) {
+function* createPageSession({ id, location, options, withUI = true }) {
```

Comment:
Regarding starting a new UAS session, I assume a new "page session". This is correct, and something I need to look into. It's probably not what I want. Would it be better to have a `restore session` API endpoint on the UAS? A bulk call is also going to help a lot with performance. Let's talk about this.

Furthermore, as mentioned in https://github.com/mozilla/tofino/pull/1415#issue-183066175 and https://github.com/mozilla/tofino/pull/1415#issuecomment-253852543 there is a problem currently with fetching autocompletions after a session restore, and I assume this is a generalized problem causing other issues due to creating page sessions when they do actually exist already (we're not closing the UAS when reloading the browser for example, and the page session ids are reused). I'll investigate.